### PR TITLE
Merging feature/employee-be to /develop

### PR DESCRIPTION
**Pull request by VladislavBarzykin**
- [x] Edit DependencyInjection files to add employee services
- [x] add employee interface
- [x] add employee repository
- [x] add employee DTO
- [x] add employee service
- [x] add interface for employee service

### CODE REVIEW

File: LX.StaffScheduler.Api/Controllers/EmployeesController.cs [hunk 1]
```diff
@@ -1,36 +1,105 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using LX.StaffScheduler.BLL.DependencyInjection;
+using LX.StaffScheduler.BLL.DTO;
+using LX.StaffScheduler.BLL.Services.Interfaces;
+using LX.StaffScheduler.DAL;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Templates.BlazorIdentity.Pages;
```
Comment (IvanLukyanau) [inline LX.StaffScheduler.Api/Controllers/EmployeesController.cs:6]: not sure you need to have this using, please remove unnecessary in all files

File: LX.StaffScheduler.Api/Controllers/EmployeesController.cs [hunk 2]
```diff
@@ -1,36 +1,105 @@
...
+                return BadRequest("Employee data is null ");
+            }
+            try
+            {
+                var createdEmployee = await _svc.AddAsync(employeeDTO);
+                return CreatedAtAction(nameof(GetById), new { id = createdEmployee.Id }, createdEmployee);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, $"Internal server error: {ex.Message}");
```
Comment (IvanLukyanau) [inline LX.StaffScheduler.Api/Controllers/EmployeesController.cs:53]: return Problem(), see other controller example

**Review by IvanLukyanau — CHANGES_REQUESTED** [review]

**Review by IvanLukyanau — CHANGES_REQUESTED** [review]
please sync with develop, looks like there are a lot of changes which are not synchronized

checkout develop, pull, checkout 'feature/employee-be', merge develop into current

File: LX.StaffScheduler.BLL/DTO/UserContractDTO.cs [hunk 3]
```diff
@@ -4,8 +4,8 @@ public class UserContractDTO : IDTO
     {
         public int Id { get; set; }
         public int DayWeek { get; set; }
-        public TimeOnly StartContractTime { get; set; }
-        public TimeOnly EndContractTime { get; set; }
+        public DateOnly StartContractTime { get; set; }
```
Comment (IvanLukyanau) [inline LX.StaffScheduler.BLL/DTO/UserContractDTO.cs:7]: Why did you change it to DateOnly?

File: LX.StaffScheduler.BLL/DTO/UserContractDTO.cs [hunk 4]
```diff
@@ -4,8 +4,8 @@ public class UserContractDTO : IDTO
...
+        public DateOnly EndContractTime { get; set; }
```
Comment (IvanLukyanau) [inline LX.StaffScheduler.BLL/DTO/UserContractDTO.cs:8]: Same as above

**Review by IvanLukyanau — CHANGES_REQUESTED** [review]

File: LX.StaffScheduler.BLL/DTO/UserContractDTO.cs (hunk 4 above)
Comment (VladislavBarzykin) [reply LX.StaffScheduler.BLL/DTO/UserContractDTO.cs]: Accident too
